src/ring_theory/witt_vector/truncated.lean

```diff
@@ -0,0 +1,245 @@
+/-
+Copyright (c) 2020 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin, Robert Y. Lewis
+-/
+
+import ring_theory.witt_vector.init_tail
+import tactic.equiv_rw
+
+/-!
+
+# Truncated Witt vectors
+
+The ring of truncated Witt vectors (of length `n`) is a quotient of the ring of Witt vectors.
+It retains the first `n` coefficients of each Witt vector.
+In this file, we setup the basic quotient API for this ring.
+
+The ring of Witt vectors is the projective limit of all the rings of truncated Witt vectors.
+We prove this in future work.
+
+## Main declarations
+
+- `truncated_witt_vector`: the underlying type of the ring of truncated Witt vectors
+- `witt_vector.truncate_fun`: the truncation map that truncates a Witt vector,
+                              to obtain a truncated Witt vector
+                              (in future work, this will be bundled into a homomorphism)
+- `truncated_witt_vector.comm_ring`: the ring structure on truncated Witt vectors
+
+## TODO
+
+Show that `witt_vector p R` is the projective limit of the system
+`truncated_witt_vector p n R` as `n` varies.
+
+-/
+
+open function (injective surjective)
+
+noncomputable theory
+
+variables {p : ℕ} [hp : fact p.prime] (n : ℕ) (R : Type*)
+
+local notation `𝕎` := witt_vector p -- type as `\bbW`
+
+/--
+A truncated Witt vector over `R` is a vector of elements of `R`,
+i.e., the first `n` coefficients of a Witt vector.
+We will define operations on this type that are compatible with the (untruncated) Witt
+vector operations.
+
+`truncated_witt_vector p n R` takes a parameter `p : ℕ` that is not used in the definition.
+This `p` is used to define the ring operations, and so it is needed to infer the proper ring
+structure. (`truncated_witt_vector p₁ n R` and `truncated_witt_vector p₂ n R` are definitionally
+equal but will have different ring operations.)
+-/
+@[nolint unused_arguments]
+def truncated_witt_vector (p : ℕ) (n : ℕ) (R : Type*) := fin n → R
```

Comment:
Should this include `[hp : fact p.prime]`?

Reply:
thanks, good point. I've now clarified this in the docstring

Reply:
Where is `hp` actually used?

Reply:
https://github.com/leanprover-community/mathlib/pull/5162/files#diff-9f4c72d184c4e4fa647884d7eca110d1e37265f84907396bb359c88ec3f72e14R164
You can't define the addition and multiplication without it.